Source/Plugins/Core/com.equella.core/resources/web/js/versioncheck.js

```
@@ -1,8 +1,178 @@
-function checkVersion(mmr, commit, display, versionUrl, aurl, cb)
-{
-    $.ajax(versionUrl + "?mmr=" + encodeURIComponent(mmr) + "&commit=" + encodeURIComponent(commit)
-    + "&display=" + encodeURIComponent(display) +"&aurl=" + encodeURIComponent(aurl)).done(function (data){
-        cb(data.newer, data.url);
+const releaseCheckUrl =
+  "https://api.github.com/repos/openequella/openEQUELLA/releases";
+const semanticVersionPattern = new RegExp(/^(\d+?)\.(\d+?)\.(\d+?)$/);
+
+function checkVersion(currentVersion, callback) {
+  $.ajax(releaseCheckUrl).done(function(data) {
+    const checkResult = createCheckResult(currentVersion, data);
+    callback(checkResult.newer, JSON.stringify(checkResult.newerReleases));
+  });
+}
+
+function createCheckResult(currentVersion, data) {
+  let newerReleaseFound = false;
+  let newerReleases = null;
+  const currentVersionMatched = currentVersion.match(semanticVersionPattern);
+  if (currentVersionMatched) {
```

Should we maybe have an else here rather than failing silently?